src/Raffle/CachingMeetupService.php

```diff
@@ -0,0 +1,73 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Raffle;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+final class CachingMeetupService implements MeetupServiceInterface
+{
+    private const ONE_HOUR = 3600;
+
+    /**
+     * @var MeetupServiceInterface
+     */
+    private $inner;
+
+    /**
+     * @var CacheItemPoolInterface
+     */
+    private $cacheItemPool;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(
+        MeetupServiceInterface $inner,
+        CacheItemPoolInterface $cacheItemPool,
+        RequestStack $requestStack
+    ) {
+        $this->inner = $inner;
+        $this->cacheItemPool = $cacheItemPool;
+        $this->requestStack = $requestStack;
+    }
+
+    public function getPresentAndPastEvents(): array
+    {
+        if ($this->requestStack->getMasterRequest()->query->get('cache_busting', false)) {
+            return $this->inner->getPresentAndPastEvents();
```

Comment:
I love the decorator here! It indeed makes caching and getting the actual results a separate concern.
One thing I noticed: this does not have the exact same behaviour. Previously, using the `cache_busting=true` query parameter would mean that the results were gotten from the source and the cache would be overridden. That last part doesn't seem to happen here, and that would not make it a "cache buster" anymore

Reply:
Nice catch! Fixed :slightly_smiling_face: